docs: rewrite creating-a-repository-system-session.md to STE rules (#2034)
apache/maven-resolver: src/site/markdown/creating-a-repository-system-session.md

    }
```

Only the local repository must be specified. The other settings have default values.

Read the API documentation for `RepositorySystemSession.SessionBuilder` to learn about all the settings for a session.

If you use a Maven plugin or run code embedded in Maven, the session is already created for you. You can derive a new session with the copy constructor of `DefaultRepositorySystemSession` if you must alter the session.

> To read configuration from the user's `settings.xml` in a non-Maven project, use the [MIMA](https://github.com/maveniverse/mima) library. MIMA provides the necessary parts. Direct any questions about MIMA to the Maven mailing list.

Reviewer: The `[Apache Maven](http://maven.apache.org/)` hyperlink from the original was dropped. The PR description states "Preserved the code example and all links verbatim," but this link is missing. STE governs language simplification, not navigational markup — please restore the link.

Also, the original's "seek a closer cooperation with Apache Maven" was broader than the rewrite's "in a non-Maven project" — the original implies any project wanting tighter Maven integration, not only non-Maven projects.

```suggestion
To integrate more closely with [Apache Maven](http://maven.apache.org/) and read configuration from the user's `settings.xml`, use the [MIMA](https://github.com/maveniverse/mima) library. MIMA provides the necessary parts. Direct any questions about MIMA to the Maven mailing list.
```

Author: This was a deliberate human rewrite following the initial LLM edit.
